GliaWidgets/Sources/CallVisualizer/VideoCall/Coordinator/VideoCallCoordinator.swift

```diff
@@ -0,0 +1,92 @@
+import UIKit
+
+extension CallVisualizer {
+    final class VideoCallCoodinator: FlowCoordinator {
+        typealias ViewController = UIViewController
+        private let theme: Theme
+
+        var delegate: ((DelegateEvent) -> Void)?
+        let environment: Environment
+        let call: Call
+        let uiConfig: RemoteConfiguration?
+        var viewModel: VideoCallViewModel?
+        var viewController: VideoCallViewController?
+
+        init(
+            theme: Theme = Theme(),
+            environment: Environment,
+            uiConfig: RemoteConfiguration?,
+            call: Call
+        ) {
+            self.environment = environment
+            self.theme = theme
+            self.uiConfig = uiConfig
+            self.call = call
+        }
+
+        func start() -> ViewController {
+            return showVideoCallViewController()
+        }
+
+        func resume() -> ViewController {
+            return resumeVideoCallViewController()
+        }
+
+        private func showVideoCallViewController() -> ViewController {
+            typealias Props = VideoCallViewController.Props
+            if let uiConfig {
+                theme.applyRemoteConfiguration(uiConfig, assetsBuilder: .standard)
+            }
+
+            let viewModel = VideoCallViewModel(
+                style: theme.callStyle,
+                environment: .init(
+                    data: environment.data,
+                    uuid: environment.uuid,
+                    gcd: environment.gcd,
+                    imageViewCache: environment.imageViewCache,
+                    timerProviding: environment.timerProviding,
+                    uiApplication: environment.uiApplication,
+                    date: environment.date,
+                    engagedOperator: environment.engagedOperator,
+                    screenShareHandler: environment.screenShareHandler
+                ),
+                call: call
+            )
+            self.viewModel = viewModel
+
+            let viewController = VideoCallViewController(props: viewModel.makeProps())
+            viewController.modalPresentationStyle = .overFullScreen
+            self.viewController = viewController
+
+            viewModel.delegate = { [weak self, weak viewController] event in
+                switch event {
+                case let .propsUpdated(props):
+                    viewController?.props = props
+                case .minimized:
+                    self?.delegate?(.close)
+                }
+            }
+            return viewController
+        }
+
+        private func resumeVideoCallViewController() -> ViewController {
+            guard let viewModel = viewModel, let viewController = viewController else { return showVideoCallViewController() }
+
+            if let uiConfig {
```

Comment:
This shorthand will not work for Xcode 13 with older version for Swift compiler, probably.

Reply:
fixed [54b8b08](https://github.com/salemove/ios-sdk-widgets/pull/494/commits/54b8b08379b18922451d03af2c4e497bac940b1a)